cocos/animation/marionette/pose-graph/foundation/pose-graph-node.ts

```diff
@@ -0,0 +1,67 @@
+import { EditorExtendable, assertIsTrue } from '../../../../core';
+import { EnterNodeInfo } from './authoring/enter-node-info';
+import type { PoseGraphNodeShell } from './node-shell';
+
+export const shellTag = Symbol('Shell');
+
+/**
+ * @zh
+ * 姿势图中的结点类。
+ * @en
+ * Class of node in pose graph.
+ */
+export class PoseGraphNode extends EditorExtendable {
+    /**
+     * @zh
+     * 该结点在姿势图中的外壳。
+     * 在结点添加到姿势图中时，该字段被置为相应的外壳对象；
+     * 当结点被移出姿势图时，该字段被置空。
+     * @en
+     * The node's shell in pose graph.
+     * This field is set to the corresponding shell object when this node is added into pose graph,
+     * is set to `undefined` when this node is removed from pose graph.
+     * @internal
+     */
+    get [shellTag] () {
+        return this._shell;
```

Comment:
The node has its shell referenced here since no matter in runtime or during authoring, the binding system needs to be able obtain the shell of a node. Now they can easily do it throught `PoseGraphNode[shellTag]`.

I know your question. You mean that if the shell is imperceptible to node, then node should not have the reference there. I once thought about this too. If we want to complete decouple node from shell, we have to work in another way, that's: let the graph holding the relationship between nodes and shells. In further, all methods query bindings should also requiring a "belonging pose graph object" passed to them. For example: querying a binding currently can be accomplished by `getInputBinding(node: PoseNode)`. If we change, the method should be changed as `getInputBinding(poseGraph: PoseGraph, node: PoseNode)`. And its implementation should be adjusted as something like:

```ts
function getInputBinding(poseGraph: PoseGraph, node: PoseNode) {
    const shell = poseGraph.getShell(node);
    // Return bindings recorded in shell...
}
```

How do you feel on this?

Reply:
Now it's changed to be non-intrusive: https://github.com/cocos/cocos-engine/pull/14952/commits/31291aa8b9a7a62d1181621d4ae746a474b0b8a7